instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -103,8 +103,10 @@ public boolean shouldStart(Context parentContext, REQUEST request) {
     SpanKind spanKind = spanKindExtractor.extract(request);
     switch (spanKind) {
       case SERVER:
+        suppressed = ServerSpan.exists(parentContext);
```

Comment:
I was thinking along the lines of your proposal here #3734 to get rid of this switch alltogether. 

Here's my latest attempt:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/ec92357ebb1edfdce6e011954dfae7348e909425/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java#L102

https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3691/files#diff-abeb4b4ae8d648931084077221aa9f1ca4c6599941dfc9891e0d5f9cb8e7cf09


Then instrumentation type defines suppression strategies for different kinds. It also eliminates the need of Client/Server/Consumer spans